builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -360,6 +361,23 @@ func (p *puller) CacheKey(ctx context.Context, index int) (string, bool, error)
 	return k, true, nil
 }
 
+func (p *puller) getRef(ctx context.Context, diffIDs []layer.DiffID, opts ...cache.RefOption) (cache.ImmutableRef, error) {
+	var parent cache.ImmutableRef
+	if len(diffIDs) > 1 {
+		var err error
+		parent, err = p.getRef(ctx, diffIDs[:len(diffIDs)-1], opts...)
+		if err != nil {
+			return nil, err
+		}
+		defer parent.Release(context.TODO())
+	}
+	return p.is.CacheAccessor.GetByBlob(context.TODO(), ocispec.Descriptor{
+		Annotations: map[string]string{
+			"containerd.io/uncompressed": diffIDs[len(diffIDs)-1].String(),
```

Comment:
Are these formalised somewhere (and if so, is there a constant for `containerd.io/uncompressed`) ?

I see there's some other annotations that are used, and wondering if these are defined by _buildkit_ or by _containerd_. If they're defined by _buildkit_ - should they use a `buildkit` (or `io.buildkit` if we have that domain) prefix? https://github.com/moby/moby/blob/b3be2802d417d05813d6140b5bc177043288db24/builder/builder-next/adapters/localinlinecache/inlinecache.go#L106-L112

Reply:
it is defined in containerd and there is no constant. I do not own `buildkit.io`

Reply:
This is currently our recommended way to use this value. Each package can define the constant but we wanted to avoid importing packages just for a constant. This value can have a different form when used as a key or filter.